src/main/java/org/jabref/gui/EntryFromIDViewModel.java

```diff
@@ -0,0 +1,228 @@
+                }
+            }
+            return bibEntry;
+        }
+    }
+
+    public void runFetcherWorker() {
+        searchSuccesfulProperty.set(false);
+        fetcherWorker.run();
+        fetcherWorker.setOnFailed(event -> {
+            Throwable exception = fetcherWorker.getException();
+            String fetcherExceptionMessage = exception.getMessage();
+            String fetcher = fetcherName;
+            String searchId = idText.getValue();
+            if (exception instanceof FetcherException) {
+                dialogService.showErrorDialogAndWait(Localization.lang("Error"), Localization.lang("Error while fetching from %0", fetcher + "." + "\n" + fetcherExceptionMessage));
+            } else {
+                dialogService.showErrorDialogAndWait(Localization.lang("No files found.", Localization.lang("Fetcher '%0' did not find an entry for id '%1'.", fetcher, searchId) + "\n" + fetcherExceptionMessage));
+            }
+            LOGGER.error(String.format("Exception during fetching when using fetcher '%s' with entry id '%s'.", searchId, fetcher), exception);
+
+            searchingProperty.set(false);
+
+            fetcherWorker = new FetcherWorker();
+
+        });
+
+        fetcherWorker.setOnSucceeded(evt -> {
+            Optional<BibEntry> result = fetcherWorker.getValue();
+            if (result.isPresent()) {
+                final BibEntry entry = result.get();
+                Optional<BibEntry> duplicate = new DuplicateCheck(Globals.entryTypesManager).containsDuplicate(basePanel.getDatabase(), entry, basePanel.getBibDatabaseContext().getMode());
+                if ((duplicate.isPresent())) {
+                    DuplicateResolverDialog dialog = new DuplicateResolverDialog(entry, duplicate.get(), DuplicateResolverDialog.DuplicateResolverType.IMPORT_CHECK, basePanel.getBibDatabaseContext());
+                    switch (dialog.showAndWait().orElse(DuplicateResolverDialog.DuplicateResolverResult.BREAK)) {
+                        case KEEP_LEFT:
+                            basePanel.getDatabase().removeEntry(duplicate.get());
+                            basePanel.getDatabase().insertEntry(entry);
+                            break;
+                        case KEEP_BOTH:
+                            basePanel.getDatabase().insertEntry(entry);
+                            break;
+                        case KEEP_MERGE:
+                            basePanel.getDatabase().removeEntry(duplicate.get());
+                            basePanel.getDatabase().insertEntry(dialog.getMergedEntry());
+                            break;
+                        default:
+                            // Do nothing
+                            break;
+                    }
+                } else {
+                    // Regenerate CiteKey of imported BibEntry
+                    new BibtexKeyGenerator(basePanel.getBibDatabaseContext(), prefs.getBibtexKeyPatternPreferences()).generateAndSetKey(entry);
```

Comment:
The cleanup for the bibtex key should be used here - don't we have such a cleanupper? (Reason: consistency with other JAbRef Code).